### PR TITLE
Phishing model

### PR DESCRIPTION
**Pull request by erezh31**

### CODE REVIEW

File: Playbooks/playbook-DBotCreatePhishingClassifier.yml [hunk 1]
```diff
@@ -0,0 +1,365 @@
+id: DBotCreatePhishingClassifier
+version: -1
+name: DBotCreatePhishingClassifier
```
Comment (meirwah) [inline Playbooks/playbook-DBotCreatePhishingClassifier.yml]: I think playbook names are usually with spaces in our OOB content, @anara123 WDYT?

File: Playbooks/playbook-DBotCreatePhishingClassifierJob.yml [hunk 2]
```diff
@@ -0,0 +1,189 @@
+id: DBotCreatePhishingClassifierJob
+version: -1
+name: DBotCreatePhishingClassifierJob
```
Comment (meirwah) [inline Playbooks/playbook-DBotCreatePhishingClassifierJob.yml]: name with spaces?

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  description: F1 score (0-1)
+  type: number
+- contextPath: DBotPredictPhishingEvaluation.Size
+  description: Test data size
+  type: number
+scripttarget: 0
+timeout: 6µs
+runonce: true
+dockerimage: demisto/machine-learning
+runas: DBotWeakRole
```
Comment (meirwah) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: not needed right?

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+- contextPath: DBotPredictPhishingEvaluation.F1
+  description: F1 score (0-1)
+  type: number
+- contextPath: DBotPredictPhishingEvaluation.Size
+  description: Test data size
+  type: number
+scripttarget: 0
+timeout: 6µs
+runonce: true
+dockerimage: demisto/machine-learning
```
Comment (meirwah) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: don't prefer to use `demisto/dl:1.0` for this?

File: Scripts/script-DBotPredictPhishingLabel.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+comment: |-
+  Predict text label using pre-trained machine learning phishing model.
+  Notice you have to run training playbook successfully first.
+enabled: true
+args:
+- name: warningThreshold
+  description: If the number of words in the input  text smaller then threshold a
+    warning will be displayed.
+  defaultValue: "30"
+- name: modelListName
```
Comment (meirwah) [inline Scripts/script-DBotPredictPhishingLabel.yml]: where is the default value for this?

File: Scripts/script-DBotPredictPhishingLabel.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  required: true
+  description: Demisto list name that stores the machine learning model
+- name: hashData
+  auto: PREDEFINED
+  predefined:
+  - "yes"
+  - "no"
+  description: Is the input text is hashed
+  defaultValue: "no"
+- name: emailSubject
```
Comment (meirwah) [inline Scripts/script-DBotPredictPhishingLabel.yml]: where is the default value for this?

File: Scripts/script-DBotPredictPhishingLabel.yml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  description: The email subject.
+- name: emailBody
```
Comment (meirwah) [inline Scripts/script-DBotPredictPhishingLabel.yml]: where is the default value for this?

File: Scripts/script-DBotPredictPhishingLabel.yml [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+- contextPath: DBotPredictPhishingLabel.InputTextNumberOfTokens
+  description: Number of tokens in the input text
+  type: number
+- contextPath: DBotPredictPhishingLabel.InputTextTokens
+  description: The input text
+  type: string
+scripttarget: 0
+runonce: false
+dockerimage: demisto/dl:1.1
+runas: DBotWeakRole
```
Comment (meirwah) [inline Scripts/script-DBotPredictPhishingLabel.yml]: not needed right?

File: Scripts/script-DBotPredictTextLabel.yml [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+- ml
+- phishing
+comment: |-
+  Predict text label using pre-trained machine learning phishing model.
+  Notice you have to run training playbook successfully first.
+enabled: true
+args:
+- name: inputText
+  required: true
+  description: 'The input text for '
```
Comment (meirwah) [inline Scripts/script-DBotPredictTextLabel.yml]: The input text to predict on

**Comment by erezh31** [comment]
https://demo2.demisto.com/#/WorkPlan/585372

**Comment by erezh31** [comment]
@meirwah plz finish your review and let me know
@hirshbergshachar will probably do a review only next week

File: Scripts/script-DBotPreparePhishingData.yml [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+      return labels_map
+
+  def get_incidents(query, size, is_context_data_needed):
+      res = demisto.executeCommand("getIncidents", {"query": query, "size": size, "sort": "created"})
+      if res[0]['Type'] == entryTypes['error']:
+          return_error(str(res[0]['Contents']))
+      incident_list = res[0]['Contents']['data'] or []
+
+      for i in incident_list:
+          custom_fields = i.get('CustomFields', {}) or {}
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: Please add a comment here to explain that this is done to flatten the custom fields in the incident

File: Scripts/script-DBotPreparePhishingData.yml [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+  if len(incidents) == MAX_INCIDENTS:
+      demisto.log('Number of results returned from incidentsQuery has reached the limit (maxNumberOfIncidents=%d). Consider increase maxNumberOfIncidents to train on more data.' % MAX_INCIDENTS)
+  for i in incidents:
+      try:
+          text = (demisto.dt(i, EMAIL_TEXT_DT) or "")
+          subject = (demisto.dt(i, EMAIL_SUBJECT_DT) or "")
+          if not text and not subject:
+              skipped_empty += 1
+              continue
+          original_label = canonize_label(demisto.dt(i, TAG_DT))
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: What does `demisto.dt` returns if it doesn't find the item? (If it's `None` we would have an error in `canonize_label)

File: Scripts/script-DBotPreparePhishingData.yml [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+              continue
+          original_label = PHISHING_LABELS.get(original_label) if demisto.args()['phishingLabels'] != ALL_LABELS else original_label
+          if isinstance(text, basestring) and isinstance(subject, basestring) and isinstance(original_label, basestring):
+              full_txt = "%s %s" % (subject, text)
+              labels_value_counts[original_label] = labels_value_counts.get(original_label, 0) + 1
+              data_texts.append(full_txt)
+              data_labels.append(original_label)
+      except Exception:
+          skipped_exception += 1
+          continue
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: Don't we want to log the error?

File: Scripts/script-DBotPreparePhishingData.yml [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+  def store_data_as_list(data, listname):
+      data = zlib.compress(data)
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: I would wrap with try except

File: Scripts/script-DBotPreparePhishingData.yml [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+      res = demisto.executeCommand("createList", {"listName": listname,"listData": base64.encodestring(data)})
+      if isError(res[0]):
+          demisto.results(res)
+          return
+      demisto.log("Store file successfully in list " + listname)
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: Stored

File: Scripts/script-DBotPreparePhishingData.yml [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+  if skipped_missing_labels > 0:
+      demisto.log("%d incidents skipped due to missing label." % skipped_missing_labels)
+  if skipped_invalid_label > 0:
+      skipped_labels_str = ",".join(set(skipped_labels))
+      demisto.log("%d incidents skipped due to label was that not defined in phishingLabels value. Skipped labels:\n%s" % (skipped_invalid_label, skipped_labels_str))
+  if skipped_exception > 0:
+      demisto.log("%d incidents skipped due to exception in parsing" % skipped_exception)
+
+  if len(data) == 0:
+      # raise Exception("There is no data to train")
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: Remove commented out code

File: Scripts/script-DBotPreparePhishingData.yml [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+      return_error("There is no data to train")
+      sys.exit()
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: status code?

File: Scripts/script-DBotPreparePhishingData.yml [hunk 17]
```diff
@@ -0,0 +1,219 @@
...
+  demisto.results({
+      'Type': entryTypes['note'],
+      'Contents': data,
+      'ContentsFormat': formats['json'],
+      'HumanReadable': tableToMarkdown('Training Data', labels_value_counts_hr),
+      'HumanReadableFormat': formats['markdown']
+  })
+
+  file_lines = map(lambda x: x['txt'] + ' %s%s' % (TAG_PREFIX, x['label']), data)
+  txt_to_write = "\r\n".join(file_lines)
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: Why do we use windows new line?

File: Scripts/script-DBotTrainTextClassifier.yml [hunk 18]
```diff
@@ -0,0 +1,120 @@
...
+      file_names = TRAINING_FILE_NAME
+  else:
+      file_names = TRAINING_FILE_NAME.split(",")
+      file_names = [f for f in file_names if f]
+  file_paths = map(get_file_path, file_names)
+  training_file_path = combine_files(file_paths)
+
+  # evaluate model
+  train_amount, (test_amount, precision, recall) = demisto_ml.get_model_precision(training_file_path)
+  demisto.log("Model evaluation trained on %d samples and tested on %d samples:\r\nPrecision:%.2f%%\r\nRecall:%.2f%%" % (train_amount, test_amount, precision*100, recall*100))
```
Comment (hirshbergshachar) [inline Scripts/script-DBotTrainTextClassifier.yml]: Why don't we use `\n`?

File: Scripts/script-DBotTrainTextClassifier.yml [hunk 19]
```diff
@@ -0,0 +1,120 @@
...
+  total_amount = train_amount + test_amount
+  if precision <= TEST_ABOVE_THRESHOLD:
+      return_error("Model evaluation is below %.2f threshold, do not store model" % TEST_ABOVE_THRESHOLD)
+
+  # train model
+  model = demisto_ml.train_text_model(training_file_path, True)
```
Comment (hirshbergshachar) [inline Scripts/script-DBotTrainTextClassifier.yml]: demisto_ml can't return an error? If yes - surround with try except

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 20]
```diff
@@ -0,0 +1,169 @@
...
+
+  entries = demisto.executeCommand('DBotPreparePhishingData', demisto.args())
+  if isError(entries[0]):
+      demisto.results(entries[0]['Contents'])
+      sys.exit(0)
+  else:
+      entries = [e for e in entries if e['ContentsFormat'] == formats['json']]
+      labeled_data = entries[0]['Contents']
+  if len(labeled_data) == 0:
+      return_error("There is not data to evaluate")
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: `there is no`

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 21]
```diff
@@ -0,0 +1,169 @@
...
+  max_number_of_incidents = int(demisto.args()['maxNumberOfIncidents'])
+  if len(labeled_data) == max_number_of_incidents:
+      demisto.log('Results from incidentsQuery has reached to the maximum (maxNumberOfIncidents=%d). Consider increase maxNumberOfIncidents to get better results' % max_number_of_incidents)
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: `increasing`

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 22]
```diff
@@ -0,0 +1,169 @@
...
+  demisto.log("Evaluate %d samples" % len(labeled_data))
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: `Evaluating`

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 23]
```diff
@@ -0,0 +1,169 @@
...
+
+  predictions = res[0]['Contents']
+
+  df = pd.DataFrame.from_dict(predictions)
+  df.loc[:, 'Original'] = pd.Series(map(lambda x: x['label'], labeled_data))
+
+  result = []
+  for i in xrange(60, 100, 5):
+      if i < 80:
+          if i % 10 == 0:
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: Why do we skip the 5?

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 24]
```diff
@@ -0,0 +1,169 @@
...
+      return_error(res[0]['Contents'])
+
+
+  predictions = res[0]['Contents']
+
+  df = pd.DataFrame.from_dict(predictions)
+  df.loc[:, 'Original'] = pd.Series(map(lambda x: x['label'], labeled_data))
+
+  result = []
+  for i in xrange(60, 100, 5):
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: I think the logic here can be simplified, please take a look

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 25]
```diff
@@ -0,0 +1,169 @@
...
+  description: Incident key to extract email text
+  defaultValue: details
+- name: emailSubjectKey
+  description: Incident key to extract email subject
+  defaultValue: name
+- name: tagKey
+  description: Incident key to extract tag
+  defaultValue: closeReason
+- name: phishingLabels
+  description: 'Comma-separated values of email tags values and mapping. The script
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingEvaluation.yml]: Make sure to update after @kirbles19

File: Scripts/script-DBotPredictPhishingLabel.yml [hunk 26]
```diff
@@ -0,0 +1,71 @@
...
+name: DBotPredictPhishingLabel
+script: |-
+  import json
+  args = demisto.args()
+  args['inputText'] = args.get('emailSubject','') + ' ' + args.get('emailBody','')
+  res = demisto.executeCommand('DBotPredictTextLabel', demisto.args())
+  res = res[-1]
+  old_context_key_prefix = 'DBotPredictTextLabel'
+  new_context_key_prefix = 'DBotPredictPhishingLabel'
+  ec = res.get('EntryContext', {}) or {}
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingLabel.yml]: `or {}` is redundant here

File: Scripts/script-DBotPredictPhishingLabel.yml [hunk 27]
```diff
@@ -0,0 +1,71 @@
...
+  for key in ec:
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictPhishingLabel.yml]: nice

File: Scripts/script-DBotPredictTextLabel.yml [hunk 28]
```diff
@@ -0,0 +1,125 @@
...
+        "HumanReadableFormat" : formats["markdown"]
+  })
+
+
+type: python
+tags:
+- ml
+- phishing
+comment: |-
+  Predict text label using pre-trained machine learning phishing model.
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictTextLabel.yml]: We can phrase this better, please check with @kirbles19

File: Scripts/script-DBotPredictTextLabel.yml [hunk 29]
```diff
@@ -0,0 +1,125 @@
...
+  import sys, base64, re
+  import demisto_ml
+
+  THRESHOLD = int(demisto.args().get('warning_threshold', 50))
+  HASH_WORDS = (demisto.args()['hashData'] == 'yes')
+  HASH_SEED = 5381
+
+  def get_input_text_prediction(model, input_text):
+      number_of_tokens = len(input_text.split(" "))
+      about_threshold = True
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictTextLabel.yml]: @erezh31 what do we used `about_threshold` for..?

File: Scripts/script-DBotPredictTextLabel.yml [hunk 30]
```diff
@@ -0,0 +1,125 @@
...
+      if number_of_tokens < THRESHOLD:
+          about_threshold = False
+
+      label, probability =  demisto_ml.predict(model, input_text)
+      return {
+          'Label': label,
+          'Probability': "%.2f" % probability,
+          'AboveThreshold': len(input_text) >= THRESHOLD,
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPredictTextLabel.yml]: Maybe it should be here?

File: Playbooks/playbook-DBotCreatePhishingClassifier.yml [hunk 31]
```diff
@@ -0,0 +1,365 @@
...
+      scriptName: DBotTrainTextClassifier
+      type: regular
+      iscommand: false
+      brand: ""
+    nexttasks:
+      '#none#':
+      - "9"
+    scriptarguments:
+      modelStoreListName:
+        simple: ${inputs.ModelListStoreName}
```
Comment (hirshbergshachar) [inline Playbooks/playbook-DBotCreatePhishingClassifier.yml]: Change arg name to start with lower case

File: Scripts/script-DBotPredictPhishingLabel.yml (hunk 26 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingLabel.yml]: it's not believe me :), sometimes it's return explicit None cuz it's JSON parse

File: Scripts/script-DBotPredictPhishingEvaluation.yml (hunk 25 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: We gonna re-document it and create also zendesk article for that

File: Scripts/script-DBotPredictPhishingEvaluation.yml (hunk 23 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: it's too much. I will split it to 2 ranges... will be simple to read

File: Scripts/script-DBotPredictPhishingEvaluation.yml (hunk 24 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: I will split it to 2 ranges... will be simple to read

File: Scripts/script-DBotPreparePhishingData.yml (hunk 16 above)
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: removed, retrun_error already exit

File: Scripts/script-DBotPreparePhishingData.yml (hunk 15 above)
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: Done :)

File: Scripts/script-DBotPreparePhishingData.yml (hunk 12 above)
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: added

File: Scripts/script-DBotPreparePhishingData.yml (hunk 11 above)
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: handled it now

File: Scripts/script-DBotPreparePhishingData.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: it's not, can be explicit None

File: Scripts/script-DBotPreparePhishingData.yml [hunk 32]
```diff
@@ -0,0 +1,219 @@
...
+              labels_map[splited[0].strip()] = canonize_label(splited[1])
+          else:
+              labels_map[v] = canonize_label(v)
+      return labels_map
+
+  def get_incidents(query, size, is_context_data_needed):
+      res = demisto.executeCommand("getIncidents", {"query": query, "size": size, "sort": "created"})
+      if res[0]['Type'] == entryTypes['error']:
+          return_error(str(res[0]['Contents']))
+      incident_list = res[0]['Contents']['data'] or []
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: done, used `get`

File: Scripts/script-DBotPreparePhishingData.yml [hunk 33]
```diff
@@ -0,0 +1,219 @@
...
+  description: Incident key to extract email text
+  defaultValue: name
+- name: emailSubjectKey
+  description: Incident key to extract email subject
+  defaultValue: name
+- name: tagKey
+  description: Incident key to extract email tag
+  defaultValue: closeReason
+- name: phishingLabels
+  description: 'Comma-separated values of email tags values and mapping (or "*" value
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: same as before

File: Scripts/script-DBotPreparePhishingData.yml [hunk 34]
```diff
@@ -0,0 +1,219 @@
...
+
+  def canonize_label(label):
+      return label.lower().strip().replace(" ","_")
+
+  def get_phishing_map_labels(comma_values):
+      if comma_values == ALL_LABELS:
+          return comma_values
+      comma_values = comma_values.lower()
+      values = comma_values.split(",")
+      labels_map = {}
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: ok

File: Scripts/script-DBotPreparePhishingData.yml [hunk 35]
```diff
@@ -0,0 +1,219 @@
...
+  import demisto_ml
+
+  TRAIN_FILE_NAME = str(uuid.uuid4()) + ".train"
+  ALL_LABELS = "*"
+  TAG_PREFIX = demisto_ml.DEFAULT_LABEL_PREFIX
+
+  def get_context(incident_id):
+      res = demisto.executeCommand("getContext", {'id': incident_id});
+      try:
+        return res[0]['Contents']['context']
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: done

File: Scripts/script-DBotPreparePhishingData.yml [hunk 36]
```diff
@@ -0,0 +1,219 @@
...
+script: |+
+  import base64, uuid, re, zlib
+  import demisto_ml
+
+  TRAIN_FILE_NAME = str(uuid.uuid4()) + ".train"
+  ALL_LABELS = "*"
+  TAG_PREFIX = demisto_ml.DEFAULT_LABEL_PREFIX
+
+  def get_context(incident_id):
+      res = demisto.executeCommand("getContext", {'id': incident_id});
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: done

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 37]
```diff
@@ -0,0 +1,169 @@
...
+          'Threshold': threshold,
+          'Precision': '%.2f%%' % precision_score,
+          'Recall': '%.2f%%' % recall_score,
+          'Below Threshold': '%.2f%%' % precantage_skipped,
+          'F1 Score': '%.2f%%' % f1_score,
+      }
+
+  entries = demisto.executeCommand('DBotPreparePhishingData', demisto.args())
+  if isError(entries[0]):
+      demisto.results(entries[0]['Contents'])
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: done

File: Scripts/script-DBotPredictPhishingEvaluation.yml [hunk 38]
```diff
@@ -0,0 +1,169 @@
...
+  import sys
+
+  # disable warnings
+  import warnings
+  warnings.filterwarnings('always')
+
+  def get_threshold_precision(df, threshold, predicted_col, original_col, threshold_col):
+      skipped_count = len(df[df[threshold_col] < threshold])
+      if skipped_count == len(df):
+          precantage_skipped = 100
```
[the remark this answers is not in the document]
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: done

File: Scripts/script-DBotPredictTextLabel.yml (hunk 29 above)
Comment (erezh31) [reply Scripts/script-DBotPredictTextLabel.yml]: removed

File: Scripts/script-DBotPredictPhishingLabel.yml (hunk 8 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingLabel.yml]: not really but all the new automation come with this now, it's like leave it emty

File: Scripts/script-DBotPredictPhishingLabel.yml (hunk 7 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingLabel.yml]: We don't have, I think we can't use `DT` expression here, only const value

File: Scripts/script-DBotPredictPhishingLabel.yml (hunk 6 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingLabel.yml]: We don't have, I think we can't use DT expression here, only const value

File: Scripts/script-DBotPredictPhishingLabel.yml (hunk 5 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingLabel.yml]: I prefer leave this empty, and there the customer must explicit set. It cuz a bug in orange which was hard to detect

File: Scripts/script-DBotPredictPhishingEvaluation.yml (hunk 4 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: changed

File: Scripts/script-DBotPredictPhishingEvaluation.yml (hunk 3 above)
Comment (erezh31) [reply Scripts/script-DBotPredictPhishingEvaluation.yml]: same comment as before, will remove

File: Playbooks/playbook-DBotCreatePhishingClassifierJob.yml (hunk 2 above)
Comment (erezh31) [reply Playbooks/playbook-DBotCreatePhishingClassifierJob.yml]: ok

File: Playbooks/playbook-DBotCreatePhishingClassifier.yml (hunk 1 above)
Comment (erezh31) [reply Playbooks/playbook-DBotCreatePhishingClassifier.yml]: ok

**Comment by meirwah** [comment]
build fails:
image

**Comment by erezh31** [comment]
@meirwah build ok now
@hirshbergshachar @meirwah plz review, comments fixed

File: Scripts/script-DBotPreparePhishingData.yml [hunk 39]
```diff
@@ -98,16 +100,25 @@ script: |+
...
-      res = demisto.executeCommand("createList", {"listName": listname,"listData": base64.encodestring(data)})
-      if isError(res[0]):
-          demisto.results(res)
-          return
-      demisto.log("Store file successfully in list " + listname)
+      try:
+          data = zlib.compress(data)
+          res = demisto.executeCommand("createList", {"listName": listname,"listData": base64.encodestring(data)})
+          if isError(res[0]):
+              demisto.results(res[0])
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: return_error?

File: Scripts/script-DBotPreparePhishingData.yml [hunk 40]
```diff
@@ -98,16 +100,25 @@ script: |+
...
+              return
```
Comment (hirshbergshachar) [inline Scripts/script-DBotPreparePhishingData.yml]: return_error does this already right?

Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: no, return error do sys.exit, not need here

File: Scripts/script-DBotPreparePhishingData.yml (hunk 39 above)
Comment (erezh31) [reply Scripts/script-DBotPreparePhishingData.yml]: no, return error do sys.exit, not need here

**Pull request merged**